diff --git a/src/xml/lv_xml_style.c b/src/xml/lv_xml_style.c
--- a/src/xml/lv_xml_style.c
+++ b/src/xml/lv_xml_style.c
@@ -252,8 +252,8 @@ void lv_xml_style_apply(lv_obj_t * obj, const char * name, const char * value)
         else SET_STYLE_IF(blend_mode, lv_xml_blend_mode_to_enum(value));
         else SET_STYLE_IF(transform_width, lv_xml_atoi(value));
         else SET_STYLE_IF(transform_height, lv_xml_atoi(value));
-        else SET_STYLE_IF(translate_x, lv_xml_atoi(value));
-        else SET_STYLE_IF(translate_y, lv_xml_atoi(value));
+        else SET_STYLE_IF(translate_x, lv_xml_to_size(value));
+        else SET_STYLE_IF(translate_y, lv_xml_to_size(value));
         else SET_STYLE_IF(translate_radial, lv_xml_atoi(value));
         else SET_STYLE_IF(transform_scale_x, lv_xml_atoi(value));
         else SET_STYLE_IF(transform_scale_y, lv_xml_atoi(value));
